### PR TITLE
Implement pirateweather handler.

### LINKED ISSUE #7804: [Integration]: OpenWeatherMap
### Is there an existing integration?

- [X] I have searched the existing integrations.

### Use Case

* Integrating a weather data provider ensures users have easy and immediate access to real-time and historical weather data.
* Users won’t need to search, clean, and import data separately; it streamlines the process, saving time and effort, avoiding potential pitfalls by using data from a reputable weather provider that is reliable and accurate.

### Motivation

Integrating a weather data provider into mindsdb will help users to develop more precise and robust prediction models. Weather, a dynamic and impactful variable, influences patterns and outcomes across various sectors including agriculture, energy, retail, and transportation. By offering seamless access to real-time and historical weather data, users can enrich their datasets, harnessing detailed, accurate, and diverse meteorological insights to refine their algorithms. This integration not only streamlines the model development process, eliminating the cumbersome task of data sourcing and preparation, but also amplifies the predictive accuracy.

### Implementation

- [ ] Create a class inheriting from [`APIHandler`](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/api_handler.py#L150).
- [ ] Implement and test `connect`, `check_connection`, `native_query`, and `call_application_api()` methods.
- [ ] Add `__about__.py` and other boilerplate code

### Anything else?

Openweathermap exposes an API that is free for up to 1000 requests a day: 
* https://openweathermap.org/price
Potential alternatives are:
* https://open-meteo.com/ (Free for non-commercial use)
* http://pirateweather.net/en/latest/API/ (Open-source)

I am happy to work on this. Please assign me if you believe that this integration could contribute to MindsDB.

### PR DESCRIPTION
**Pull request by libklein**
## Description

Implements a handler for the PirateWeather API. 

Features:
- Historical daily forecast
- Historical hourly forecast

Fixes #7804

## Type of change

- [x] ⚡ New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Can you please add initial unittests? As a reference check https://github.com/mindsdb/mindsdb/blob/staging/tests/handler_tests/test_binance_handler.py

**Comment by libklein** [comment]
> Can you please add initial unittests? As a reference check https://github.com/mindsdb/mindsdb/blob/staging/tests/handler_tests/test_binance_handler.py

I already have unit tests set up, but did not want to publish them on Github to avoid compromising my API key. Do you have a suggestion on how to deal with the credentials issue?

**Comment by libklein** [comment]
I have read the CLA Document and I hereby sign the CLA

File: mindsdb/integrations/handlers/pirateweather_handler/README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+* `time` - Date for which to fetch historical data. Optional, defaults to the current date.
+* `units` - Units to use for temperature and wind speed. Optional, defaults to `us` (Imperial units). Other options are:
+    * `ca`: SI, with Wind Speed and Wind Gust in kilometres per hour.
+    * `uk`: SI, with Wind Speed and Wind Gust in miles per hour and visibility are in miles.
+    * `us`: Imperial units
+    * `si`: SI units
+
+## Example Usage
+
+The first step is to create a database with the new `github` engine.
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/pirateweather_handler/README.md:65]: Change engine to pirateweather

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
CAn you please share screenshots or short video of working integration?

**Comment by libklein** [comment]
> CAn you please share screenshots or short video of working integration?

Sure! Here you go:
https://github.com/mindsdb/mindsdb/assets/42714034/f51ef9c2-f48a-4ce7-b714-c84ca7bc3ea1

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**